Algorithm/QCAlgorithm.cs

```diff
@@ -539,6 +548,35 @@ public void SetParameters(Dictionary<string, string> parameters)
         }
 
         /// <summary>
+        /// Set the available data feeds in the <see cref="SecurityManager"/>
+        /// </summary>
+        /// <param name="availableDataFeeds">The different <see cref="TickType"/> each <see cref="Security"/> supports</param>
+        public void SetAvailableDataFeeds(Dictionary<SecurityType, List<TickType>> availableDataFeeds)
+        {
+            foreach (var dataFeed in availableDataFeeds)
+            {
+                AvailableDataFeeds[dataFeed.Key] = dataFeed.Value;
+            }
+        }
+
+        /// <summary>
+        /// Hard code the set of default available data feeds
+        /// </summary>
+        public Dictionary<SecurityType, List<TickType>> DefaultDataFeeds()
```

Comment:
Ultimately, with this PR, I'm trying to make it so that the 'Reader' method used by the ISubscriptionDataSourceReader can be configured by the user. The problem I found with moving the idea of the 'default data feed' outside the scope of the Algorithm is that everything hinges on the SubscriptionDataConfig, which is created within the scope of the algorithm.  It's the Type of the SubscriptionDataConfig that determines which Reader method to use.

While technically still in the scope of the algorithm, perhaps SubscriptionManager is a better place to hang the 'default data feeds'?